tools/ci-scripts/check-aws-sdk-services

```diff
@@ -7,9 +7,14 @@
 set -eux
 cd aws-sdk
 
-# Invoking `cargo test` at the root directory implicitly checks for the validity
-# of the top-level `Cargo.toml`
-cargo test --all-features
+# Check if the $ENABLE_SMOKETESTS environment variable is set
+if [ -n "$ENABLE_SMOKETESTS" ]; then
```

Comment:
Will we ever want to run these smoketests in our CI check? My understanding was that we wanted to keep all network calls out of that portion of our testing? This came up when I contributed the wasm crate because I initially put the tests in CI and was asked to move them to the Canary.

Reply:
Ref: https://github.com/smithy-lang/smithy-rs/pull/3409#discussion_r1489802497

Reply:
We run these checks in Catapult too IIRC. I wanted to use the same script in both cases, but differ in whether smoketests are enabled.

Reply:
Related to an env var `ENABLE_SMOKETESTS`, it fails when the script runs in a container
```
./smithy-rs/tools/ci-scripts/check-aws-sdk-services: line 11: ENABLE_SMOKETESTS: unbound variable
```
we need to make sure that the script runs fine in our relase pipeline (and compiling smoke tests)

Reply:
I'll remove the script change for now so that we can get this merged. We can integrate it in a separate PR